tests/envs/test_action_dim_check.py

```diff
@@ -17,3 +33,59 @@ def test_serialize_deserialize(environment_id):
 
     with pytest.raises(ValueError, match="Action dimension mismatch"):
         env.step(0.1)
+
+
+@pytest.mark.parametrize("env", make_envs_by_action_space_type(spec_list, Discrete))
+def test_discrete_actions_out_of_bound(env):
+    # Car racing currently allow `out-of-bound` actions
+    if "CarRacing" in env.spec.id:
+        return
+
+    env.reset()
+
+    action_space = env.action_space
+    upper_bound = action_space.start + action_space.n - 1
+
+    with pytest.raises(Exception):
```

Comment:
Yes, you are right

Reply:
Some have this checks:
https://github.com/openai/gym/blob/745e7059e7754e26cfcaa82d0d9a0defec3a08c4/gym/envs/box2d/lunar_lander.py#L425-L427
Some let the step fail with different errors (Acrobot will fail here with an IndexError):
https://github.com/openai/gym/blob/745e7059e7754e26cfcaa82d0d9a0defec3a08c4/gym/envs/classic_control/acrobot.py#L190
Taxy will for example fail here with a KerError:
https://github.com/openai/gym/blob/745e7059e7754e26cfcaa82d0d9a0defec3a08c4/gym/envs/toy_text/taxi.py#L210-L211
I think it is reasonable to add more consistency across error handling but I think it may be another PR

Reply:
Ok, lets do that as part of another PR